### PR TITLE
added support of simple_rnn, gru and lstm for the deepspeech module

### PR DESCRIPTION
**Pull request by EgorLakomkin**
I have added an option to select which type of the rnn to use in the DeepSpeech module - Simple RNN, GRU or LSTM. What do you think of it @SeanNaren ? With batch norm simple rnn shows suprisingly good performance and faster to train.

### CODE REVIEW

**Comment by SeanNaren** [comment]
I like the idea a lot but I think it can be done simpler, by taking the `nn.LSTM` function as a parameter:

```python
    def __init__(self, input_size, hidden_size, rnn=nn.LSTM, bidirectional=False, batch_norm=True):
        super(BatchLSTM, self).__init__()
        self.input_size = input_size
        self.hidden_size = hidden_size
        self.batch_norm_activate = batch_norm
        self.bidirectional = bidirectional
        self.batch_norm = SequenceWise(nn.Sequential(
            nn.Linear(input_size, hidden_size, bias=False),
            nn.BatchNorm1d(hidden_size)))
        self.rnn = rnn(input_size=hidden_size if batch_norm else input_size, hidden_size=hidden_size,
                       bidirectional=bidirectional, bias=False)
        self.num_directions = 2 if bidirectional else 1
```
What do you think?

**Comment by EgorLakomkin** [comment]
Yeah, you are right, we can just pass the rnn class for simplicity. I corrected the code a bit

**Comment by SeanNaren** [comment]
Thanks! I've done some small changes and one large bug fix that I'm going to slide into this as well, seems like somewhere the skip input branch got half merged into the code adding an additional linear layer that definitely shouldn't have been in the architecture!

**Comment by SeanNaren** [comment]
Thanks a lot :)
